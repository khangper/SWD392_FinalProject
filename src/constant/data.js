export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
