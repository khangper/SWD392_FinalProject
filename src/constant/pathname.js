export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
