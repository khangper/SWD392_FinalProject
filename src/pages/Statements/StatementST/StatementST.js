import React, { useState, useEffect } from 'react';
import dropdown from "../../../assets/dropdown.png";
import { useDispatch, useSelector } from "react-redux";
import { fetchStatementSTRequest } from '../../../redux/reduxActions/StatementSTAction';

const StatementST = () => {
    const dispatch = useDispatch();
    const { statement_course } = useSelector((state) => state.statementST);

    useEffect(() => {
        dispatch(fetchStatementSTRequest());
    }, [dispatch]);

    const [isOpen, setIsOpen] = useState(false);
    const [selectedValue, setSelectedValue] = useState('Monthly Invoices');

    const toggleDropdown = () => setIsOpen(!isOpen);
    const handleSelect = (value) => {
        setSelectedValue(value);
        setIsOpen(false);
    };

    return (
        <div className='statementis'>
            <div className="statementis-row">
                <div className="statementis-col-lg-12">
                    <h2 className="st_title"><i className="uil uil-file-alt"></i> Statements</h2>
                </div>
            </div>
            <div className="statementis-row">
                <div className="statementis-col-lg-8">
                    <div className="top_countries mt-30">
                        <div className="top_countries_title">
                            <h2>Earnings</h2>
                        </div>
                        <div className="statement_content">
                            <p className="tt-body">If you are not an instructor, you can't use this section.</p>
                            <a href="#" className="apply_instructor_btn">Become an Instructor</a>
                        </div>
                    </div>
                </div>

                <div className="statementis-col-lg-4">
                    <div className="top_countries mt-30">
                        <div className="top_countries_title">
                            <h2>View Invoices</h2>
                        </div>
                        <div className="statement_invoice_content">
                            <div className="date_selector">
                                <div className="ui selection" tabIndex="0" onClick={toggleDropdown}>
                                    <input name="date" type="hidden" value={selectedValue} />
                                    <div className='sis-drop'>
                                        <span className="section-title-wrapper">
                                            <span className="section-title-text">{selectedValue}</span>
                                            <img className="sidebar-dropdown-caret-icon" src={dropdown} alt="Dropdown Icon" />
                                        </span>
                                    </div>
                                    {isOpen && (
                                        <div className="dropdown-menu">
                                            <div className="top" onClick={() => handleSelect('April 2020')}>
                                                <div className="title">April 2020</div>
                                            </div>
                                            <div className="top" onClick={() => handleSelect('March 2020')}>
                                                <div className="title">March 2020</div>
                                            </div>
                                        </div>
                                    )}
                                </div>
                                <button className="st_download_btn"><i className="uil uil-download-alt"></i></button>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="statementis-col-lg-12">
                    <div className="more_options_tt">
                        <li>
                            <div>
                                <button className="more_items_14 active">This Month</button>
                            </div>
                        </li>
                        <li>
                            <div>
                                <button className="more_items_14">Last Month</button>
                            </div>
                        </li>
                        <li>
                            <div className="explore_search">
                                <div className="sis-search-box">
                                    <div className="sis-box-input">
                                        <input className="sis-search" type="text" placeholder="Document Number" />
                                        <i className="uil uil-search-alt icon icon8"></i>
                                    </div>
                                </div>
                            </div>
                        </li>
                    </div>
                </div>
            </div>
            <div className="MyCourse-tab-pane active" id="my-courses-tab" role="tabpanel">
                <div className="MyCouser-container">
                    <div className="MyCouser-table-container">
                        <table className="MyCourse-table ucp-table">
                            <thead className="MyCourse-thead-s">
                                <tr>
                                    <th className="Statement-th-text" scope="col">Date</th>
                                    <th className="Statement-th-text">Order ID</th>
                                    <th className="Statement-th-text" scope="col">Type</th>
                                    <th className="Statement-th-text" scope="col">Title</th>
                                    <th className="Statement-th-text" scope="col">Amount</th>
                                    <th className="Statement-th-text" scope="col">Invoice</th>
                                </tr>
                            </thead>
                            <tbody>
                                {statement_course.map(course => (
                                    <tr key={course.id}>
                                        <td className="MyCourse-text-center">{course.date}</td>
                                        <td className="MyCourse-text-center">{course.orderID}</td>
                                        <td className="MyCourse-text-center">{course.type}</td>
                                        <td className="MyCourse-text-center">{course.title}</td>
                                        <td className="MyCourse-text-center">{course.amount}</td>
                                        <td className="MyCourse-text-center">
                                            <a href="#">View</a>
                                        </td>
                                    </tr>
                                ))}
                            </tbody>
                        </table>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default StatementST;
