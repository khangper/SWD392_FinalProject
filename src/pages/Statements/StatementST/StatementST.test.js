import React from "react";
import { renderWithProviders } from "../../../constant/test-util";
import StatementST from "./StatementST"
describe("StatementST", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the ReportHistory component", async () => {
        const { container } = renderWithProviders(<StatementST />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });
})