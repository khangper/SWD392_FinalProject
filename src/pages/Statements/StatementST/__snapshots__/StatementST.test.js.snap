// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword renders the ReportHistory component 1`] = `
<div>
  <div
    class="statementis"
  >
    <div
      class="statementis-row"
    >
      <div
        class="statementis-col-lg-12"
      >
        <h2
          class="st_title"
        >
          <i
            class="uil uil-file-alt"
          />
           Statements
        </h2>
      </div>
    </div>
    <div
      class="statementis-row"
    >
      <div
        class="statementis-col-lg-8"
      >
        <div
          class="top_countries mt-30"
        >
          <div
            class="top_countries_title"
          >
            <h2>
              Earnings
            </h2>
          </div>
          <div
            class="statement_content"
          >
            <p
              class="tt-body"
            >
              If you are not an instructor, you can't use this section.
            </p>
            <a
              class="apply_instructor_btn"
              href="#"
            >
              Become an Instructor
            </a>
          </div>
        </div>
      </div>
      <div
        class="statementis-col-lg-4"
      >
        <div
          class="top_countries mt-30"
        >
          <div
            class="top_countries_title"
          >
            <h2>
              View Invoices
            </h2>
          </div>
          <div
            class="statement_invoice_content"
          >
            <div
              class="date_selector"
            >
              <div
                class="ui selection"
                tabindex="0"
              >
                <input
                  name="date"
                  type="hidden"
                  value="Monthly Invoices"
                />
                <div
                  class="sis-drop"
                >
                  <span
                    class="section-title-wrapper"
                  >
                    <span
                      class="section-title-text"
                    >
                      Monthly Invoices
                    </span>
                    <img
                      alt="Dropdown Icon"
                      class="sidebar-dropdown-caret-icon"
                      src="dropdown.png"
                    />
                  </span>
                </div>
              </div>
              <button
                class="st_download_btn"
              >
                <i
                  class="uil uil-download-alt"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="statementis-col-lg-12"
      >
        <div
          class="more_options_tt"
        >
          <li>
            <div>
              <button
                class="more_items_14 active"
              >
                This Month
              </button>
            </div>
          </li>
          <li>
            <div>
              <button
                class="more_items_14"
              >
                Last Month
              </button>
            </div>
          </li>
          <li>
            <div
              class="explore_search"
            >
              <div
                class="sis-search-box"
              >
                <div
                  class="sis-box-input"
                >
                  <input
                    class="sis-search"
                    placeholder="Document Number"
                    type="text"
                  />
                  <i
                    class="uil uil-search-alt icon icon8"
                  />
                </div>
              </div>
            </div>
          </li>
        </div>
      </div>
    </div>
    <div
      class="MyCourse-tab-pane active"
      id="my-courses-tab"
      role="tabpanel"
    >
      <div
        class="MyCouser-container"
      >
        <div
          class="MyCouser-table-container"
        >
          <table
            class="MyCourse-table ucp-table"
          >
            <thead
              class="MyCourse-thead-s"
            >
              <tr>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Date
                </th>
                <th
                  class="MyCourse-text-center"
                >
                  Order ID
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Type
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Title
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Amount
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Invoice
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`StatementST renders the ReportHistory component 1`] = `
<div>
  <div
    class="statementis"
  >
    <div
      class="statementis-row"
    >
      <div
        class="statementis-col-lg-12"
      >
        <h2
          class="st_title"
        >
          <i
            class="uil uil-file-alt"
          />
           Statements
        </h2>
      </div>
    </div>
    <div
      class="statementis-row"
    >
      <div
        class="statementis-col-lg-8"
      >
        <div
          class="top_countries mt-30"
        >
          <div
            class="top_countries_title"
          >
            <h2>
              Earnings
            </h2>
          </div>
          <div
            class="statement_content"
          >
            <p
              class="tt-body"
            >
              If you are not an instructor, you can't use this section.
            </p>
            <a
              class="apply_instructor_btn"
              href="#"
            >
              Become an Instructor
            </a>
          </div>
        </div>
      </div>
      <div
        class="statementis-col-lg-4"
      >
        <div
          class="top_countries mt-30"
        >
          <div
            class="top_countries_title"
          >
            <h2>
              View Invoices
            </h2>
          </div>
          <div
            class="statement_invoice_content"
          >
            <div
              class="date_selector"
            >
              <div
                class="ui selection"
                tabindex="0"
              >
                <input
                  name="date"
                  type="hidden"
                  value="Monthly Invoices"
                />
                <div
                  class="sis-drop"
                >
                  <span
                    class="section-title-wrapper"
                  >
                    <span
                      class="section-title-text"
                    >
                      Monthly Invoices
                    </span>
                    <img
                      alt="Dropdown Icon"
                      class="sidebar-dropdown-caret-icon"
                      src="dropdown.png"
                    />
                  </span>
                </div>
              </div>
              <button
                class="st_download_btn"
              >
                <i
                  class="uil uil-download-alt"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="statementis-col-lg-12"
      >
        <div
          class="more_options_tt"
        >
          <li>
            <div>
              <button
                class="more_items_14 active"
              >
                This Month
              </button>
            </div>
          </li>
          <li>
            <div>
              <button
                class="more_items_14"
              >
                Last Month
              </button>
            </div>
          </li>
          <li>
            <div
              class="explore_search"
            >
              <div
                class="sis-search-box"
              >
                <div
                  class="sis-box-input"
                >
                  <input
                    class="sis-search"
                    placeholder="Document Number"
                    type="text"
                  />
                  <i
                    class="uil uil-search-alt icon icon8"
                  />
                </div>
              </div>
            </div>
          </li>
        </div>
      </div>
    </div>
    <div
      class="MyCourse-tab-pane active"
      id="my-courses-tab"
      role="tabpanel"
    >
      <div
        class="MyCouser-container"
      >
        <div
          class="MyCouser-table-container"
        >
          <table
            class="MyCourse-table ucp-table"
          >
            <thead
              class="MyCourse-thead-s"
            >
              <tr>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Date
                </th>
                <th
                  class="MyCourse-text-center"
                >
                  Order ID
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Type
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Title
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Amount
                </th>
                <th
                  class="MyCourse-text-center"
                  scope="col"
                >
                  Invoice
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
